test/valid-data/type-intarray/main.ts

```diff
@@ -0,0 +1 @@
+export type MyType = Uint8Array;
```

Comment:
Instead of using a Uint8Array in the test (I still think it makes no sense to create a schema for this type) you could test only the actual problematic code:

```typescript
interface Types {
    String: string;
}

export type MyObject = Types[keyof Types];
```

This produces the same error which is fixed by this PR and the schema simply looks like this:

```
{
  "$ref": "#/definitions/MyObject",
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "MyObject": {
      "type": "string"
    }
}
```

And maybe a better name for the test is then `type-indexed-access-keyof`